File: benchmarks/xlang/src/Collatz.java
public class Collatz {
    static long steps(long n) {
        long c = 0;
        while (n != 1) { n = n % 2 == 0 ? n / 2 : 3 * n + 1; c++; }
        return c;
    }
    public static void main(String[] a) {
        long t0 = System.nanoTime();
        long total = 0;
        for (long i = 1; i <= 300000; i++) total += steps(i);
        System.out.println("CHECK " + total);
        System.out.println("MS " + (System.nanoTime() - t0) / 1000000);
    }
}

File: benchmarks/xlang/src/collatz.lua
local function steps(n)
  local c = 0
  while n ~= 1 do
    if n % 2 == 0 then n = n // 2 else n = 3 * n + 1 end
    c = c + 1
  end
  return c
end
local t0 = os.clock()
local total = 0
for i = 1, 300000 do total = total + steps(i) end
print("CHECK " .. total)
print("MS " .. math.floor((os.clock() - t0) * 1000))
